src/polynomial.h

```diff
@@ -86,6 +86,8 @@ class Polynomial {
     void add(Polynomial<T>* b);
     void sub(Polynomial<T>* b);
     void mul(Polynomial<T>* b);
+    void mul(Polynomial<T>* b, int deg_out);
```

Comment:
Commit should explain why we need these specialization/how they improve the perf.

Reply:
agree